src/util/sandbox/fd_sandbox.c

```diff
@@ -206,140 +156,369 @@ switch_user( uint uid, uint gid ) {
+  };
 
-  FD_TESTV( !prctl( PR_SET_KEEPCAPS, 1 ) );
-  switch_user( uid, gid );
+  uint ids[] = {
+    uid_in_parent,
+    gid_in_parent
+  };
 
-  struct __user_cap_header_struct capheader;
-  capheader.version = _LINUX_CAPABILITY_VERSION_3;
-  capheader.pid = 0;
-  struct __user_cap_data_struct capdata[2] = { {0} };
-  FD_TESTV( !syscall( SYS_capget, &capheader, capdata ) );
-  capdata[ CAP_TO_INDEX( CAP_SYS_ADMIN ) ].effective |= CAP_TO_MASK( CAP_SYS_ADMIN );
-  FD_TESTV( !syscall( SYS_capset, &capheader, capdata ) );
+  for( ulong i=0UL; i<2UL; i++ ) {
+    int fd = open( MAP_PATHS[ i ], O_WRONLY );
+    if( -1==fd ) FD_LOG_ERR(( "open(%s) failed (%i-%s)", MAP_PATHS[ i ], errno, fd_io_strerror( errno ) ));
 
-  FD_TESTV( !unshare( CLONE_NEWUSER | CLONE_NEWNS | CLONE_NEWNET | CLONE_NEWCGROUP | CLONE_NEWIPC | CLONE_NEWUTS ) );
-  FD_TESTV( !prctl( PR_SET_KEEPCAPS, 0 ) );
-  deny_setgroups();
-  userns_map( uid, "uid_map" );
-  userns_map( gid, "gid_map" );
+    char map_line[ 64 ];
+    FD_TEST( fd_cstr_printf_check( map_line, sizeof( map_line ), NULL, "1 %u 1\n", ids[ i ] ) );
+    long written = write( fd, map_line, strlen( map_line ) );
+    if( -1L==written ) FD_LOG_ERR(( "write(%s) failed (%i-%s)", MAP_PATHS[ i ], errno, fd_io_strerror( errno ) ));
+    if( written != (long)strlen( map_line ) ) FD_LOG_ERR(( "write(%s) failed to write all data", MAP_PATHS[ i ] ));
+    if( close( fd ) ) FD_LOG_ERR(( "close(%s) failed (%i-%s)", MAP_PATHS[ i ], errno, fd_io_strerror( errno ) ));
+  }
+}
+
+void
+fd_sandbox_private_deny_namespaces( void ) {
+  static char const * SYSCTLS[] = {
+    "/proc/sys/user/max_user_namespaces",
+    "/proc/sys/user/max_mnt_namespaces",
+    "/proc/sys/user/max_cgroup_namespaces",
+    "/proc/sys/user/max_ipc_namespaces",
+    "/proc/sys/user/max_net_namespaces",
+    "/proc/sys/user/max_pid_namespaces",
+    "/proc/sys/user/max_uts_namespaces",
+  };
+
+  static char * VALUES[] = {
+    "1", /* One user namespace is allowed, to created the nested child. */
+    "1", /* One mount namespace is allowed, to pivot the root in the user namespace */
+    "0",
+    "0",
+    "0",
+    "0",
+    "0",
+  };
 
-  FD_TESTV( !prctl( PR_SET_DUMPABLE, 0, 0, 0, 0 ) );
-  for ( int cap = 0; cap <= CAP_LAST_CAP; cap++ ) {
-    FD_TESTV( !prctl( PR_CAPBSET_DROP, cap, 0, 0, 0 ) );
+  for( ulong i=0UL; i<sizeof(SYSCTLS)/sizeof(SYSCTLS[ 0 ]); i++) {
+    int fd = open( SYSCTLS[ i ], O_WRONLY );
+    if( FD_UNLIKELY( fd<0 ) ) FD_LOG_ERR(( "open(%s) failed (%i-%s)", SYSCTLS[ i ], errno, fd_io_strerror( errno ) ));
+    long written = write( fd, VALUES[ i ], 1 );
+    if( FD_UNLIKELY( written==-1 ) ) FD_LOG_ERR(( "write(%s) failed (%i-%s)", SYSCTLS[ i ], errno, fd_io_strerror( errno ) ));
+    else if( FD_UNLIKELY( written!=1 ) ) FD_LOG_ERR(( "write(%s) failed to write data", SYSCTLS[ i ] ));
+    if( FD_UNLIKELY( close( fd ) ) ) FD_LOG_ERR(( "close(%s) failed (%i-%s)", SYSCTLS[ i ], errno, fd_io_strerror( errno ) ));
```

Comment:
This is pretty similar to above, but maybe it's not worth factoring it out for just two uses